etl/grapher_helpers.py

```diff
@@ -499,3 +500,8 @@ def from_values(xs: List[int]) -> "IntRange":
 
     def to_values(self) -> list[int]:
         return [self.min, self.max]
+
+
+def contains_inf(s: pd.Series) -> bool:
+    """Check if a series contains infinity."""
+    return pd.api.types.is_numeric_dtype(s.dtype) and np.isinf(s).any()  # type: ignore
```

Comment:
Isn't the `np.isinf(s)` going to raise a TypeError if the series is not numeric? If not, all good.

Reply:
It won't get past the first condition. If it's not numeric then the second part after `and` won't be evaluated.

(Another way would be to catch `TypeError`)